src/main/java/pl/onsight/wypozyczalnia/controller/AdminController.java

```diff
@@ -32,29 +35,64 @@ public ModelAndView adminPage(ModelAndView modelAndView) {
         return modelAndView;
     }
 
+    @DeleteMapping(path = "/product-order/{id}")
+    @ResponseBody
+    public void removeProductOrder(@PathVariable Long id) {
+        productOrderService.removeProductOrder(id);
+    }
+
+    @DeleteMapping(path = "/product/{id}")
+    @ResponseBody
+    public void removeProduct(@PathVariable Long id) {
+        productService.removeProduct(id);
+    }
+
+
     @PostMapping("/admin-page")
     public ModelAndView addProduct(@ModelAttribute @Valid ProductEntity product,
                                    BindingResult bindingResult, ModelAndView modelAndView) {
+        List<ProductEntity> allProducts = productService.findAllProducts();
+        List<String> allNamesOfProducts = allProducts.stream().map(name -> name.getProductName()).collect(Collectors.toList());
         if (bindingResult.hasErrors()) {
             modelAndView.setViewName("admin");
+        } else if (allNamesOfProducts.contains(product.getProductName())) {
+            modelAndView.addObject("info", new Info("Produkt o nazwie " + product.getProductName() + " już istnieje!", false));
+            return adminPage(modelAndView);
         } else {
             productService.addProduct(product);
             modelAndView.setViewName("redirect:/shop");
         }
         return modelAndView;
     }
 
-    @DeleteMapping(path = "/product-order/{id}")
-    @ResponseBody
-    public void removeProductOrder(@PathVariable Long id) {
-        productOrderService.removeProductOrder(id);
+    @GetMapping("/edit/{id}")
+    public ModelAndView showUpdateForm(@PathVariable("id") Long id, ModelAndView modelAndView) {
+        ProductEntity product = productService.findProductById(id);
+        modelAndView.addObject("product", product);
+        modelAndView.setViewName("updateProduct");
+        return modelAndView;
     }
 
-    @DeleteMapping(path = "/product/{id}")
-    @ResponseBody
-    public void removeProduct(@PathVariable Long id) {
-        productService.removeProduct(id);
+    @PostMapping("/update/{id}")
+    public ModelAndView updateProduct(@PathVariable("id") Long id, @Valid ProductEntity product,
+                                      BindingResult result, ModelAndView modelAndView) {
+        List<ProductEntity> allProducts = productService.findAllProducts();
+        List<String> allNamesOfProducts = allProducts.stream().map(name -> name.getProductName()).collect(Collectors.toList());
+        String tempName=product.getProductName();
+        allNamesOfProducts.remove(tempName);
+        String newName=productService.findProductById(id).getProductName();
+        if (result.hasErrors()) {
+            product.setId(id);
+            modelAndView.setViewName("admin");
+            return modelAndView;
```

Comment:
To tak średnio wygląda zwraca tylko modelAndView z ustawionym tylko view na admin. Troche tego nie czaje musze zobaczyć  u Ciebie jak to działa

Reply:
no to trzeba poprawic, bo nie dziala :). i w walidatorze powinno sprawdzac :)